src/pybind/mgr/volumes/fs/stats_util.py

```diff
@@ -157,36 +167,50 @@ def __init__(self, vc, volname, dst_svname, src_path, dst_path):
         # bug when it occurs. Therefore, it needs to be fixed. The issue with
         # code in/underlying class RTimer.
         self.update_task = RTimer(1, self.update)
-        self.update_task.start()
+
+    def add_clone(self, volname, dst_sv_name, src_path, dst_path):
+        self.clones.append(CloneInfo(volname, dst_sv_name, src_path, dst_path))
+
+        if not self.update_task.is_alive() and len(self.clones) > 0:
+            self.update_task.start()
 
     def update(self):
         assert self.pev_id is not None
+        total_clones = len(self.clones)
+        assert total_clones > 0
+        percent = 0.0
 
-        with open_volume(self.vc, self.volname) as fsh:
-            _, _, percent = get_amount_copied(self.src_path, self.dst_path,
+        for clone in self.clones:
+            with open_volume_lockless(self.vc, clone.volname) as fsh:
+                percent += get_percent_copied(clone.src_path, clone.dst_path,
                                               fsh)
 
+        percent = round(percent / total_clones, 3)
         # progress module takes progress as a fraction between 0.0 to 1.0.
         progress_fraction = percent / 100
-        msg = f'Subvolume "{self.dst_svname}" has been {percent}% cloned'
+        msg = f'Avg progress made by {total_clones} clones is {percent}%'
 
         self.vc.mgr.remote('progress', 'update', ev_id=self.pev_id,
                            ev_msg=msg, ev_progress=progress_fraction,
                            refs=['mds', 'clone'], add_to_ceph_s=True)
 
-        if progress_fraction == 1.0:
+        if progress_fraction >= 1.0:
+            if progress_fraction > 1.0:
+                log.error('avg percentage of cloning completed is more than '
```

Comment:
Did you run into this or is this safe programming?

Reply:
Safe programming.